engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -410,13 +437,15 @@ public void writeTuple(Object obj, String port)
       storage.writeDataItem(bos.toByteArray(), true);
       //logger.debug("Writing tuple for port id {}", pi.id);
       //fsOutput.hflush();
-      ++totalTupleCount;
       if (numSubscribers > 0) {
         // this is not asynchronous.  we need to fix this
         publishTupleData(pi.id, obj);
       }
-    } catch (IOException ex) {
-      logger.error(ex.toString());
+      ++totalTupleCount;
```

Comment:
Subject to int overflow? Why it is necessary to move increment of totalTupleCount here. Should it be incremented each time writeTuple is called to count the number of attempts or it only counts successful attempts?